source/extensions/clusters/dns_srv/dns_srv_cluster.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <chrono>
+#include <cstdint>
+#include <functional>
+#include <string>
+#include <unordered_set>
+
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/config/endpoint/v3/endpoint.pb.h"
+#include "envoy/config/endpoint/v3/endpoint_components.pb.h"
+#include "envoy/extensions/clusters/dns_srv/v3/cluster.pb.h"
+#include "envoy/extensions/clusters/dns_srv/v3/cluster.pb.validate.h"
+#include "envoy/stats/scope.h"
+
+#include "source/common/common/empty_string.h"
+#include "source/common/upstream/cluster_factory_impl.h"
+#include "source/common/upstream/upstream_impl.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class DnsSrvClusterFactory;
+class DnsSrvClusterTest;
+
+/**
+ * Cluster implementation for DNS SRV records.
+ * An SRV record specifies hostname(s) for a service, each hostname additionally contains:
+ * - weight
+ * - priority
+ * - port number
+ *
+ * For each hostname, we will resolve a list of IP addresses (A/AAAA records) and populate the
+ * cluster endpoints.
+ *
+ * Current limitations:
+ * - Only c-ares DNS resolver is supported.
+ * - Only one SRV record is supported per cluster.
+ * - All resolved IP addresses will be added to cluster (acting like Strict DNS cluster).
+ * - Weight and priority are ignored.
```

Comment:
It would be great to fix this, but you don't have to in this PR if you don't want. But if you keep this limitation, please also add this to the protobuf docs (as that is the user-facing documentation).

In fact, all the limitations listed here should move to the protobuf docs.